docs/hugo/content/design/ADR-2023-09-Skipping-Properties/_index.md

```diff
@@ -0,0 +1,418 @@
+
+-->
+
+{{< figure src="./conversion-v4-v5-class.png" >}}
+
+When we convert from **v5** to **v4**, again the `ResidentialAddress` property gets serialized into the `PropertyBag` on the **v4** Person. 
+
+Again, it's useful to see a concrete example:
+
+<!-- yuml.me
+
+[v4.Person|FullName: Michael Theodore Mouse;FamilyName: Mouse; KnownAs: Mickey|Properties PropertyBag]
+[v4.Person]-.-[ResidentialAddress {bg:cornsilk}|Street: '1313 S. Harbor Blvd'\nSuburb: ''\nCity: 'Anaheim, CA 92803'\nCountry: 'USA']
+[v5.Person|FullName: Michael Theodore Mouse;FamilyName: Mouse; KnownAs: Mickey; ResidentialAddress v3.Address|Properties PropertyBag]
+[v5.Person]->[v5.Address|Street: 1313 S. Harbor Blvd\nSuburb:\nCity: Anaheim, CA 92803\nCountry: USA|Properties PropertyBag]
+
+-->
+
+{{< figure src="./conversion-v4-v5-instance.png" >}}
+
+As before, there is nowhere in the **v4** `Person` to store Mickey's residential address, so it gets safely stashed away in the `PropertyBag`.
+
+```yaml
+fullName: Michael Theodore Mouse
+familyName: Mouse
+knownAs: Mickey
+propertyBag:
+    residentialAddress: "street:1313 S. Harbor Blvd;suburb:;city:Anaheim, CA 92803;country:USA"
+```
+
+However, note that this time the bag contains a **v5** `Address` in a serialized form, **not** a **v3** `Address`.
+
+In the other direction, from **v4** to **v5**, works fine, as we can deserialize the **v5** `Address` from the property bag.
+
+Conversion back and forward between versions **v4** and **v5** works fine.
+
+### The Problem
+
+Observe how we can end up with two different variants of **v4** `Person`. In one case, we have a **v4** `Person` where the property bag contains a **v3** `Address`; in the other case, we have a **v4** `Person` the property bag contains a **v5** `Address`:
+
+<!-- yuml.me 
+
+[v4.Person|FullName: Michael Theodore Mouse;FamilyName: Mouse; KnownAs: Mickey|Properties PropertyBag]
+[v4.Person]-.-[v3.ResidentialAddress {bg:cornsilk}|Label:\n'1313 S. Harbor Blvd',\n'Anaheim',\n'CA 92803',\n'USA']
+[v4.Person]-.-[v5.ResidentialAddress {bg:cornsilk}|Street: '1313 S. Harbor Blvd'\nSuburb: ''\nCity: 'Anaheim, CA 92803'\nCountry: 'USA']
+
+-->
+
+{{< figure src="./conflict-v3-v5.png" >}}
+
+Round trips between adjacent versions work fine, but we run into problems.
+
+When we convert from **v3** to **v4** to **v5** we end up stuck part-way through.
+
+<!-- yuml.me 
+
+[v3.Person|FullName string;FamilyName string; KnownAs string; ResidentialAddress v3.Address|Properties PropertyBag]
+[v3.Person]->[v3.Address|Label multilineString|Properties PropertyBag]
+
+[v4.Person|FullName string;FamilyName string; KnownAs string|Properties PropertyBag]
+
+[v5.Person|FullName string;FamilyName string; KnownAs string;ResidentialAddress|Properties PropertyBag]
+[v5.Person]--[v5.Address|Street string; Suburb string; City string; Country string|Properties PropertyBag]
+
+-->
+
+{{< figure src="./conversion-v3-v4-v5-class.png" >}}
```

Comment:
I think in the class diagram, the some of the Address fields are missing a type specifier.